### DISCUSSION TITLE
How to reference a package(in local Monorepo) in a single frontend project？

### DISCUSSION THREAD

**Discussion by Carnia** [pnpm/pnpm#4840] [Q&A]
# Pre-content:
1. I have multiple frontend projects
2. The public parts of these front-end projects are extracted
3. Front-end projects and public parts are **not in the same git repository**

# detail:
1. Use pnpm's workspace to manage these public parts, such as **ui, utils**:
````sh
- packages
   |- - ui
   |- - utils
- pnpm-workspace.ymal
````
2. Use in front-end projects A and B
````
import { Icon } from "@myPackage/ui";
import { isEmpty } from "@myPackage/utils";
...
````
# question:
When I was developing A locally, I wanted to change the style of the Icon component in `@myPackage/ui`. Do I link `@myPackage/ui` and `@myPackage/ui` within A project?

Will this be a problem? Note that the dependency in `@myPackage/ui` is `"@myPackage/utils": "workspace:*"`.

I haven't solved this problem: how to quickly debug and develop `@myPackage/ui` in project A

**Comment by kaikaibenkai** [comment]
+1. I have the same problem!

**Comment by zkochan** [comment]
If the projects are in separate workspaces, you can only connect them with a symlink by running

```
pnpm link <relative path to the project that should be linked in>
```

(https://pnpm.io/cli/link)

**Comment by vaynevayne** [comment]
I succeeded, show it to the latecomers.
```jsx
~/workspace/my-npm/packages/core/package.json // name: @ui/core
~/workspace/business-project/package.json
```
just  do `cd business-project` then 
`pnpm link ../my-npm/packages/core` 
or 
`pnpm link ~/workspace/my-npm/packages/core` 

note: `don't add --dir`
You will see business-project's node_modules has an `@ui/core`
There is no need to do anything in the my-npm and no need to execute pnpm install.

**Reply by dominictobias-bullish** [reply]
That did nothing, at least nothing was added to my package.json
